kernel/preemption/src/lib.rs

```diff
@@ -9,26 +9,30 @@
 extern crate alloc;
 
 use core::sync::atomic::{AtomicU8, Ordering};
+use cpu::CpuId;
 
-/// Theseus uses a `u8` to hold each CPU core's ID (apic ID),
-/// so the maximum number of cores it supports is `u8::MAX` (256).
-const MAX_CPU_CORES: usize = u8::MAX as usize;
+/// Theseus previously used a `u8` to hold each CPU core's ID
+/// so the maximum number of cores it supports is 256.
+///
+/// In the future, we will use CPU-local storage for each CPU's preemption count,
+/// so we won't need to use a system-wide primitive array here.
+const MAX_CPU_CORES: usize = 2_usize.pow(u8::BITS);
```

Comment:
Why use pow? `u8::MAX` was simpler to understand

Reply:
`u8:MAX` was incorrect. It should be 256, not 255. I can just use a const instead.